### PR TITLE
Add a FreeBSD simple generic worker.

### PR DESCRIPTION
**Pull request by verm**
Add support for a simple generic worker on FreeBSD.

`build.sh -a` works fine for me with this patch and to build on FreeBSD:

  `GOOS=freebsd GOARCH=amd64 ./build.sh`

### CODE REVIEW

File: workers/generic-worker/runtime/runtime_freebsd.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// +build darwin linux freebsd
+
```
Comment (petemoore) [inline workers/generic-worker/runtime/runtime_freebsd.go]: This new file shouldn't be needed, as `OSUser` struct is already defined in `workers/generic-worker/runtime/runtime_posix.go` and has `freebsd` added as a build constraint.

File: workers/generic-worker/simple_posix.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// +build darwin,simple linux,simple
+// +build darwin,simple linux,simple,simple freebsd
```
Comment (petemoore) [inline workers/generic-worker/simple_posix.go]:
```suggestion
// +build darwin,simple linux,simple freebsd,simple
```

File: workers/generic-worker/diskspace_posix.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ func freeDiskSpaceBytes(dir string) (uint64, error) {
 		return 0, err
 	}
 	// Available blocks * size per block = available space in bytes
-	b := stat.Bavail * uint64(stat.Bsize)
+	b := uint64(stat.Bavail) * uint64(stat.Bsize)
```
Comment (petemoore) [inline workers/generic-worker/diskspace_posix.go]: What is the reason for this change?

Comment (verm) [reply workers/generic-worker/diskspace_posix.go]: Without this change I get this error:

`./diskspace_posix.go:17:19: invalid operation: stat.Bavail * uint64(stat.Bsize) (mismatched types int64 and uint64)`

File: workers/generic-worker/runtime/runtime_freebsd.go (hunk 1 above)
Comment (verm) [reply workers/generic-worker/runtime/runtime_freebsd.go]: You're right!  There must have been an artifact laying around as I went ahead and did an `rm -r`, rebuilt and got the error.  It's been a while since I've worked with Go forgive my rustiness.

File: workers/generic-worker/chain_of_trust_posix.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// +build multiuser,linux multiuser,darwin
+// +build multiuser,linux multiuser,darwin,freebsd
```
Comment (petemoore) [inline workers/generic-worker/chain_of_trust_posix.go]: This change isn't needed - chain of trust isn't supported in simple engine.

File: workers/generic-worker/diskspace_posix.go (hunk 3 above)
Comment (petemoore) [reply workers/generic-worker/diskspace_posix.go]: Fair enough, I guess this must be an inconsistency in implementation across platforms. Then this should be fine.

File: workers/generic-worker/chain_of_trust_posix.go (hunk 4 above)
Comment (verm) [reply workers/generic-worker/chain_of_trust_posix.go]: Okay thank you for the clear explanation.  I looked at my test script and wasn't building multiuser.  I've changed it now to just use `build.sh -a` now for testing before submission sorry for the trouble.
